core/shared/src/main/scala/fs2/async/Ref.scala

```diff
@@ -67,125 +34,70 @@ final class Ref[F[_],A] private[fs2] (implicit F: Effect[F], ec: ExecutionContex
    * setter first. Once it has noop'd or been used once, a setter
    * never succeeds again.
    */
-  def access: F[(A, A => F[Boolean])] =
-    F.flatMap(F.delay(new MsgId)) { mid =>
-      F.map(getStamped(mid)) { case (a, id) =>
-        val set = (a: A) =>
-          F.async[Boolean] { cb => actor ! Msg.TrySet(id, a, x => cb(Right(x))) }
-        (a, set)
-      }
-    }
-
-  override def get: F[A] = F.flatMap(F.delay(new MsgId)) { mid => F.map(getStamped(mid))(_._1) }
+  def access: F[(A, A => F[Boolean])] = F.delay {
+    def snapshot = ar.get
```

Comment:
mm it works regardless since `snapshot` is evaluated when I put it into the tuple, but `val` is clearer in any case, so I'll change it to that. Good catch!

Reply:
ah no, you're right!
`get` in `access` and `snapshot` is different. Great catch then!